Scripts/utils/print_links.py

```diff
@@ -1,6 +1,64 @@
+from typing import Iterable, Tuple
+from shapely import Point, LineString
+
 from utils.calc_noise import NoiseModel
 
 
+class GeometryType:
+    name: str
+    geom_type: str
+    def __new__(cls, obj):
+        pass
+
+
+class Node(GeometryType):
+    name = "NODE"
+    geom_type = "Point"
+
+    def __new__(cls, node):
+        return Point(node.x, node.y)
+
+
+class Link(GeometryType):
+    name = "LINK"
+    geom_type = "LineString"
+
+    def __new__(cls, link):
+        return LineString(link.shape)
+
+
+def geometries(attr_names: Iterable[str],
+               objects: Iterable,
+               geom_type: GeometryType) -> Tuple[Iterable, dict]:
```

Comment:
The geom_type argument is a class and not an instance. The type annotation should be Type[GeometryType] (or type[GeometryType] for newer python)